x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java

```diff
@@ -493,131 +549,139 @@ public void runAsDenied(Authentication authentication, String action, TransportM
         }
-        sb.append("]");
-        return sb.toString();
     }
 
-    static String subject(Authentication authentication) {
-        final StringBuilder sb = new StringBuilder("principal=[");
-        sb.append(authentication.getUser().principal()).append("], realm=[");
+    private static void subject(Authentication authentication, StringMapMessage logEntry) {
+        logEntry.with(PRINCIPAL_FIELD_NAME, authentication.getUser().principal());
         if (authentication.getUser().isRunAs()) {
-            sb.append(authentication.getLookedUpBy().getName()).append("], run_by_principal=[");
-            sb.append(authentication.getUser().authenticatedUser().principal()).append("], run_by_realm=[");
+            logEntry.with(PRINCIPAL_REALM_FIELD_NAME, authentication.getLookedUpBy().getName())
+                    .with(PRINCIPAL_RUN_BY_FIELD_NAME, authentication.getUser().authenticatedUser().principal())
+                    .with(PRINCIPAL_RUN_BY_REALM_FIELD_NAME, authentication.getAuthenticatedBy().getName());
+        } else {
+            logEntry.with(PRINCIPAL_REALM_FIELD_NAME, authentication.getAuthenticatedBy().getName());
         }
-        sb.append(authentication.getAuthenticatedBy().getName()).append("]");
-        return sb.toString();
-    }
-
-    private static String hostAttributes(RestRequest request) {
-        final InetSocketAddress socketAddress = request.getHttpChannel().getRemoteAddress();
-        String formattedAddress = NetworkAddress.format(socketAddress.getAddress());
-        return "origin_address=[" + formattedAddress + "]";
     }
 
-    protected static String originAttributes(ThreadContext threadContext, TransportMessage message, LocalNodeInfo localNodeInfo) {
-        return restOriginTag(threadContext).orElse(transportOriginTag(message).orElse(localNodeInfo.localOriginTag));
+    private static void opaqueId(ThreadContext threadContext, StringMapMessage logEntry) {
+        final String opaqueId = threadContext.getHeader(Task.X_OPAQUE_ID);
+        if (opaqueId != null) {
+            logEntry.with(OPAQUE_ID_FIELD_NAME, opaqueId);
+        }
     }
 
-    private String opaqueId() {
-        String opaqueId = threadContext.getHeader(Task.X_OPAQUE_ID);
-        if (opaqueId != null) {
-            return ", opaque_id=[" + opaqueId + "]";
-        } else {
-            return "";
+    private static void restOrigin(RestRequest request, StringMapMessage logEntry) {
+        assert LOCAL_ORIGIN_FIELD_VALUE.equals(logEntry.get(ORIGIN_TYPE_FIELD_NAME)); // this is the default
+        final InetSocketAddress socketAddress = request.getHttpChannel().getRemoteAddress();
+        if (socketAddress != null) {
+            final String formattedAddress = NetworkAddress.format(socketAddress.getAddress());
+            logEntry.with(ORIGIN_TYPE_FIELD_NAME, REST_ORIGIN_FIELD_VALUE).with(ORIGIN_ADDRESS_FIELD_NAME, formattedAddress);
         }
+        // fall through to local_node default
```

Comment:
by default all events originate locally unless one of `restOrigin` or `restOrTransportOrigin` is called to populate the origin details.